lib/basedriver/helpers.js

```diff
@@ -121,6 +124,28 @@ function verifyAppExtension (app, supportedAppExtensions) {
     supportedAppExtensions);
 }
 
+async function calculateFolderIntegrity (folderPath) {
+  return (await fs.glob('**/*', {cwd: folderPath, strict: false, nosort: true})).length;
+}
+
+async function calculateFileIntegrity (filePath) {
+  return await fs.hash(filePath);
+}
+
+async function isAppIntegrityOk (currentPath, expectedIntegrity = {}) {
+  if (!await fs.exists(currentPath)) {
+    return false;
+  }
+  const isDir = (await fs.stat(currentPath)).isDirectory();
+  if (isDir && await calculateFolderIntegrity(currentPath) >= expectedIntegrity?.folder) {
```

Comment:
so, does this mean current folder size can be greater than the cached one..?
I thought this case also basically compare cached one and current one with `===`

Reply:
Yes. this is for the case OS devices to add its own hidden caches there, for example TimeMachine or Finder ones

Reply:
ah, I see.
Maybe a comment about this will help us when we take a look around this code again in the future

Reply:
added a comment there